mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,183 @@
+    #        data[:, ch_count // 2] = np.arange(n_samples) + 100```
+    orig_data = raw_data / (raw.info["chs"][0]["cal"] * raw.info["chs"][0]["range"])
+    assert np.allclose(sum(orig_data[:, 50] - 10 - np.arange(n_channels)), 76.0)
+
+    assert np.allclose(orig_data[n_channels // 2, :100] - 100, np.arange(100))
+    assert np.allclose(orig_data[n_channels // 2, 150:] - 100, np.arange(150))
+
+    data, times = raw.get_data(start=10, stop=20, return_times=True)
+    assert n_channels, 10 == data.shape
+
+    data, times = raw.get_data(start=0, stop=300, return_times=True)
+    epochs = make_fixed_length_epochs(raw, duration=0.05, preload=False)
+    assert len(epochs.events) == 3
+    epochs = make_fixed_length_epochs(raw, duration=0.05, preload=True)
+    assert len(epochs) == 2
+    assert "BAD_ACQ_SKIP" in epochs.drop_log[1]
+
+
+@requires_testing_data
+def test_nsx_ver_22():
+    """Primary tests for NEURALCD reader."""
+    raw = read_raw_nsx(
+        nsx_22_fname,
+    )
+    assert getattr(raw, "_data", False) is False
+    assert raw.info["sfreq"] == 2000
+
+    # Check info object
+    assert raw.info["meas_date"].day == 31
+    assert raw.info["meas_date"].year == 2023
+    assert raw.info["meas_date"].month == 1
+    assert raw.info["chs"][0]["cal"] == 0.6103515625
+    assert raw.info["chs"][0]["range"] == 0.001
+
+    # check raw_extras
+    for r in raw._raw_extras:
+        assert r["orig_format"] == raw.orig_format
+        assert r["orig_nchan"] == 128
+        assert len(r["timestamp"]) == len(r["nb_data_points"])
+        assert len(r["timestamp"]) == len(r["offset_to_data_block"])
+
+    # Check annotations
+    assert len(raw.annotations) == 0
+
+    raw = _test_raw_reader(
+        read_raw_nsx,
+        input_fname=nsx_22_fname,
+        eog=None,
+        misc=None,
+        test_scaling=False,  # XXX this should be True
+        test_rank=False,
+    )
+    raw_data, times = raw[:]
+    n_channels, n_times = raw_data.shape
+    assert times.shape[0] == n_times
+    # Check data
+    # There is only one contiguous data packet, samples 0--100. Data
+    # was generated as:
+    #     ```data = np.ones((n_samples, ch_count))
+    #        data[n_samples // 2] = np.arange(ch_count) + 10
+    #        data[:, ch_count // 2] = np.arange(n_samples) + 100```
+    orig_data = raw_data / (raw.info["chs"][0]["cal"] * raw.info["chs"][0]["range"])
+    assert np.allclose(sum(orig_data[:, 50] - 10 - np.arange(n_channels)), 76.0)
+
+    assert np.allclose(orig_data[n_channels // 2, :100] - 100, np.arange(100))
+
+    data, times = raw.get_data(start=10, stop=20, return_times=True)
+    assert n_channels, 10 == data.shape
+
+    data, times = raw.get_data(start=0, stop=300, return_times=True)
+    epochs = make_fixed_length_epochs(raw, duration=0.05, preload=True, id=1)
+    assert len(epochs) == 1
+    assert epochs.event_id["1"] == 1
+    with pytest.raises(ValueError) as excinfo:
```

Comment:
See where this is done elsewhere, to me looks a lot simpler with a matchs kwarg  in pytest.raises